File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/40_교집합/main.cpp
#include<stdio.h>
#include<vector>
#include<algorithm>
using namespace std;
int main() {
	int n, m, i, p1=0, p2=0, p3=0;
	scanf("%d", &n);
	vector<int> a(n);
	for(i=0;i<n;i++){
		scanf("%d", &a[i]);
    }
	sort(a.begin(), a.end());
	scanf("%d", &m);
	vector<int> b(m), c(n+m);
	for(i=0;i<m;i++){
	    scanf("%d", &b[i]);
	}
	sort(b.begin(), b.end());
	while(p1< n && p2 < m) {
		if (a[p1] == b[p2]) {
			c[p3++] = a[p1++];
			p2++;
		} else if (a[p1] > b[p2]) {
			p2++;
		} else {
			p1++;
		}
	}
	for(i=0;i<p3;i++) {
		printf("%d ", c[i]);
    }

	return 0;
}

File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/40_교집합/main.go
package main

import (
	"fmt"
	"os"
	"sort"
)

func main() {
	file, _ := os.Open("input.txt")
	defer file.Close()
	oldStd := os.Stdin
	defer func() { os.Stdin = oldStd }()
	os.Stdin = file
	var N, M int
	fmt.Scanf("%d", &N)

	arr1 := make([]int, N)
	for i := 0; i < N; i++ {
		fmt.Scanf("%d", &arr1[i])
	}

	fmt.Scanf("%d", &M)

	arr2 := make([]int, M)
	for i := 0; i < M; i++ {
		fmt.Scanf("%d", &arr2[i])
	}

	var result []int

	var p1, p2 int

	sort.Ints(arr1)
	sort.Ints(arr2)

	for p1 < N && p2 < M {
		if arr1[p1] == arr2[p2] {
			result = append(result, arr1[p1])
			p1++
			p2++
		} else if arr1[p1] > arr2[p2] {
			p2++
		} else {
			p1++
		}
	}
	fmt.Println(result)
}
